components/yaml/introduction.rst

```diff
@@ -219,6 +222,49 @@ representation to the inline one:
         foo: bar
         bar: baz
 
+Indentation
+"""""""""""
+
+By default the YAML component will use 4 spaces for indentation. This can be
+changed using the second argument as follows::
+
+    // use 8 spaces for indentation
+    echo Yaml::dump($array, 2, 8);
+
+.. code-block:: yaml
+
+    foo: bar
+    bar:
+            foo: bar
+            bar: baz
+
+Invalid Types and Object Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default the YAML component will encode any "unsupported" type (i.e.
+resources and objects) as ``null``.
+
+Instead of encoding as ``null`` you can choose to throw an exception if an invalid
+type is encountered in either the dumper or parser as follows::
+
+    // throw an exception if a resource or object is encoutered
+    $dumper->dump($data, 2, 4, true);
+
+    // throw an exception if an encoded object is found in the YAML string
+    $parser->parse($yaml, true);
+
+However, you can activate object support using the next argument::
+
+    $object = new \stdClass();
+    $object->hello = 'goodbye';
+
+    $dumped = $dumper->dump($object, 2, 0, false, true);
```

Comment:
In the case of the `Dumper` it should be `0` - as it is called recursively and the actual indentation level is set via. a setter.